### ISSUE TITLE
Parallel should be named Promise

### ISSUE THREAD

**Issue by alexzielenski** [pointfreeco/episode-code-samples#13]
Hey guys! I just stumbled upon your content and I love the ideas you are proposing - they really get my gears turning with some new ways to think about code. I noticed watching your episode you had trouble thinking of a name for the concept of `F3` and I just wanted to point out that the `Promise` type from other languages like Javascript is eerily similar to this concept of `Parallel`, or `F3` with the only difference being that promises separate a `success` and a `failure` response into a formal construct while the parallel type guarantees a reply unless the inner type is optional. Even with this slight syntactic difference, I think the construct is identical.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

**Comment by stephencelis** [comment]
Hi @alexzielenski!

Thanks for the comments! And good intuitions here! Promise is definitely related to Parallel, but as you note, it has some additional and different semantics. In addition to baking success and failure into the type:

- Promises are eager. They execute their side effects as soon as they are instantiated. Parallel requires that someone calls `run` to kick things off. We like the laziness of Parallel because it makes them easier to compose values without accidentally executing side effects.
- Promises have a single operator, `then`, that can either transform the wrapped value or take that value and use it to kick off another promise in sequence. Parallel is a simple type that has `map` for pure transformations on its wrapped value, but it has no notion of sequencing.

For these reasons we think it's important to consider Parallel as its own type. This makes it easier to study things like `map` and `zip` alongside other types that implement `map` and `zip`. Promise abstracts its behavior differently and hides some of this stuff with its overloaded `then`.

We delayed giving it a name in order to emphasize the parallel function's shape over its semantics. When it was finally time to give it a name, we named it after a [PureScript module](https://github.com/purescript/purescript-parallel) that behaves the same way 😄

There are a lot of other type names out there in this space. In addition to Promise, there's Future, Task, IO, and more!
